docs/develop/01_sdk/02_cookbook/06_upgrading_to_v0_29/01_backwards_compatibility.md

```diff
@@ -3,32 +3,24 @@ id: v29-backwards-compatibility
 title: Backwards Compatibility / Interoperability with Previous Versions
 ---
 
-Beside the breaking code changes we have identified three data format changes which might affect interoperability.
-The didsign.io and w3n.id already accept bare requests for attestation, which are effectively the new Credentials, so nothing should break there.
+Depending on how exactly your application interacts with other applications, the changed data formats might make them incompatible.
+Supporting only the latest version will result in poor user experience and therefore is not recommended.
+
+We have only identified the Credentials API as definitely having breaking changes. 
+The new version 3.0 of [Credentials API specification](https://github.com/KILTprotocol/spec-ext-credential-api) uses the data types from the SDK version 0.29.
+Since the specification requires the dApp and the extension to announce the versions of the API they use,
+it is relatively simple to make your application backwards compatible.
+Note that we anticipate the extensions to be upgraded first, so the dApps will not have to be backwards compatible.
 
 In the attester flow the messages `submit-terms` and `request-attestation` are affected.
 In the verifier flow the `submit-credential` message is affected.
-These changes relate to the Credentials API and will require a new version of it released.
-The current specification defines the way for the extension to announce the version of specification it adheres to.
-This makes it possible for the DApp to detect the version mismatch.
-Unfortunately, the extension cannot detect the version mismatch.
-The new specification version should also require the DApp to announce the version of specification it adheres to.
-(This will be a non-enumerable property: `Object.defineProperty(window.kilt, 'meta', { value: { versions: { credentials: '3.0' } }, enumerable: false })`)
-
-With the versions known the apps might choose to either bail on version mismatch, or attempt to be compatible.
-Bailing will result in a bad user experience without recourse, so we want to avoid that.
-Backwards compatibility is not straightforward though, and we want to limit its scope.
-While we only have one extension published, there are already several DApp implementations, so we will try to handle the compatibility in the extension: the Sporran.
-This will address the case when the extension speaks the newer version, and we will try to avoid the opposite case by releasing the upgraded Sporran much earlier.
-We will not handle the case when the user manually disabled Sporran updates.
+The extension can achieve compatibility by translating the messages received from and sent to the previous versions of DApps.
+We will link here to the code examples to achieve that, but you can also implement the following steps yourself.
```

Comment:
```suggestion
We will provide below the code examples to achieve that, but you can also implement the following steps yourself.
```

It looks like the code examples are directly in this document, not linked to.

Reply:
I wanted to link to code examples in the Sporran repo, not to include the actual code in this document